server/odc-service/src/main/java/com/oceanbase/odc/service/common/DruidMonitor.java

```diff
@@ -0,0 +1,646 @@
+                    .append("jdbcCommitCount=").append(jdbcCommitCount).append(", ")
+                    .append("jdbcReadOnlyCount=").append(jdbcReadOnlyCount).append(", ")
+                    .append("jdbcRollbackCount=").append(jdbcRollbackCount).append(", ")
+                    .append("jdbcPoolConnectionOpenCount=").append(jdbcPoolConnectionOpenCount).append(", ")
+                    .append("jdbcPoolConnectionCloseCount=").append(jdbcPoolConnectionCloseCount).append(", ")
+                    .append("jdbcResultSetOpenCount=").append(jdbcResultSetOpenCount).append(", ")
+                    .append("jdbcResultSetCloseCount=").append(jdbcResultSetCloseCount).append(", ")
+                    .append("jdbcExecuteCount=").append(jdbcExecuteCount).append(", ")
+                    .append("jdbcExecuteErrorCount=").append(jdbcExecuteErrorCount).append(", ")
+                    .append("jdbcExecuteTimeMillis=").append(jdbcExecuteTimeMillis).append(", ")
+                    .append("jdbcFetchRowCount=").append(jdbcFetchRowCount).append(", ")
+                    .append("jdbcUpdateCount=").append(jdbcUpdateCount).append(", ")
+                    .append("lastError=")
+                    .append(lastError == null ? "" : StringUtils.removeWhitespace(lastError.toString())).append(", ")
+                    .append("lastErrorTime=")
+                    .append(lastErrorTime != null ? new SimpleDateFormat(DATA_FORMAT_PATTERN).format(lastErrorTime)
+                            : "null")
+                    .append(", ")
+                    .append("histogram=").append(histogram != null ? Arrays.toString(histogram) : "null")
+                    .toString();
+
+        }
+
+        public Long computeAvgExecuteTime() {
+            return executeTimeMillis / executeCount;
+        }
+
+        public Long computeAvgJdbcExecuteTime() {
+            return jdbcExecuteTimeMillis / executeCount;
+        }
+
+        public Long computeAvgJdbcExecuteCount() {
+            return jdbcExecuteCount / executeCount;
+        }
+
+        private String getMethodIdentification() {
+            return "class=" + clazz + "，method=" + method;
+        }
+
+        public String buildTooLongAvgExecuteTimeMsg() {
+            return getMethodIdentification() + ", avgExecuteTime=" + computeAvgExecuteTime() + ";";
+        }
+
+        public String buildTooLongJdbcAvgExecuteTimeMsg() {
+            return getMethodIdentification() + ", avgJdbcExecuteTime=" + computeAvgJdbcExecuteTime() + ";";
+        }
+
+        public String buildTooMuchJdbcExecuteCountMsg() {
+            return getMethodIdentification() + ", avgJdbcExecuteCount=" + computeAvgJdbcExecuteCount()
+                    + ";";
+        }
+
+        public static DruidMethodStatVO mapToVO(Map<String, Object> map) {
+            DruidMethodStatVO vo = new DruidMethodStatVO();
+            vo.setClazz((String) map.get(DruidMethodStatConstants.CLASS));
```

Comment:
will `map.get()` return null ?

Reply:
Some fields should semantically be null, whereas other fields related to statistical values have already been given default values.